doc/ipython-notebooks/multiclass/KNN.ipynb

```diff
@@ -260,14 +260,21 @@
     "knn.set_k(3)\n",
     "knn.set_knn_solver_type(KNN_COVER_TREE)\n",
     "pred = knn.apply_multiclass(feats_test)\n",
-    "print \"Covertree KNN took %2.1fs\" % (Time.get_curtime() - start)\n"
+    "print \"Covertree KNN took %2.1fs\" % (Time.get_curtime() - start)\n",
+    "\n",
+    "\n",
+    "start = Time.get_curtime()\n",
+    "knn.set_k(3)\n",
+    "knn.set_knn_solver_type(KNN_KDTREE)\n",
+    "pred = knn.apply_multiclass(feats_test)\n",
+    "print \"KDTree KNN took %2.1fs\" % (Time.get_curtime() - start)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "So we can significantly speed it up. Let's do a more systematic comparison. For that a helper function is defined to run the evaluation for KNN:"
+    "Obviously applying KNN is very costly: for each prediction you have to compare the object against all training objects. While the implementation in SHOGUN will use all available CPU cores to parallelize this computation it might still be slow when you have big data sets. We can see it been significantly speed up by Cover Trees solver, but the result of KD-Trees solver is not so ideally. So, let's do a more systematic comparison. For that a helper function is defined to run the evaluation for KNN:"
```

Comment:
We can see it been significantly speed up by Cover Trees solver, but the result of KD-Trees solver is not so ideally. 


This sentence doesnt really make sense, re-formulate